static/elements/chromedash-approvals-dialog.js

```diff
@@ -159,7 +173,14 @@ class ChromedashApprovalsDialog extends LitElement {
       (res) => {
         this.comments = res.comments;
       });
-    Promise.all([p1, p2, p3]).then(() => {
+    const p4 = window.csClient.getApprovalConfigs(this.featureId).then(
+      (res) => {
+        this.configs = res.configs;
+        this.showConfigs = new Set(this.configs.map(c => c.field_id));
+        this.changedConfigsByField = new Map();
+        this.possibleOwners = res.possible_owners;
+      });
+    Promise.all([p1, p2, p3, p4]).then(() => {
```

Comment:
IIUC, when `p1,p2,p3,p4` promises have resolved, this is resolved. What if one of them get rejected, is `this.loading` still true?

Reply:
Yes, that is true.  If any one of the individual promises is rejected, then the Promise.all() also rejects.  There is no reject case, so the user would just sit there seeing a spinner.

To improve that UX, I have added a catch() clause that shows an error message and closes the dialog.

Reply:
Yup thanks for addressing it